internal/painter/gl/texture.go

```diff
@@ -113,15 +113,11 @@ func (p *painter) newGlCircleTexture(obj fyne.CanvasObject) Texture {
 func (p *painter) newGlImageTexture(obj fyne.CanvasObject) Texture {
 	img := obj.(*canvas.Image)
 
-	width := p.textureScale(img.Size().Width)
-	height := p.textureScale(img.Size().Height)
-
-	tex := paint.PaintImage(img, p.canvas, int(width), int(height))
```

Comment:
If this really isn't needed here then can you apply the same fix to the `software/draw.go` as well?
Then I don't think this `PaintImage` method is needed any more either...

Reply:
The software backend does actually use PaintImage to do the scaling. You can see that the size being passed is actually different from the size of the image in the software case and not in the case of GL.

Reply:
Isn't it the same here? PaintImage is being given the current size of the object, not the size of the image pixels

Reply:
It isn't exactly the same, but I think you are right. I will drop that part of the change for now and come back to it later.